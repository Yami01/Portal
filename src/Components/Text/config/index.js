/**
 *
 * @format
 * @flow
 *
 */

export const FONT_REGULAR = 'regular'
export const FONT_BOLD = 'bold'
export const FONT_SEMIBOLD = 'semiBold'
export const FONT_LIGHT = 'light'
