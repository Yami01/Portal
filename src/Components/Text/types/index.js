/**
 *
 * @format
 * @flow
 *
 */
import type { Node } from "react"
import type { TextProps } from "react-native/Libraries/Text/TextProps"
import type { TextStyleProp } from "react-native/Libraries/StyleSheet/StyleSheet"

export type PropsType = {
  ...TextProps,
  fontWeight?: 'light' | 'regular' | 'semiBold' | 'bold',
  fontSize?: number,
  color?: string,
  style?: TextStyleProp,
  testID?: string,
  children?: Node,
  allowFontScaling?: boolean,
}

export type TextStylesType = {}
