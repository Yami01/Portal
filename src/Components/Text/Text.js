/**
 *
 * Text
 * @format
 * @flow
 *
 */

import type { AbstractComponent, Node } from "react"
import React, { memo, useMemo } from "react"
import { Text as LibText } from "react-native"
import type { TextStyleProp } from "react-native/Libraries/StyleSheet/StyleSheet"
import { useTheme } from "react-native-paper"
import { FONT_REGULAR } from "./config"
import type { PropsType } from "./types"

const Text: AbstractComponent<PropsType> = memo((props: PropsType): Node => {
  const {
    fontWeight = FONT_REGULAR,
    fontSize,
    color,
    style,
    children,
    testID,
    ...restProps
  } = props
  const theme = useTheme()

  const textStyle = useMemo((): TextStyleProp => {
    const finalTextStyle: TextStyleProp[] = []

    if (fontWeight) {
      finalTextStyle.push(theme.fonts[fontWeight])
    }
    if (fontSize) {
      finalTextStyle.push({ fontSize })
    }
    if (color) {
      finalTextStyle.push({ color })
    }
    finalTextStyle.push(style)

    return finalTextStyle
  }, [theme, fontSize, fontWeight, color, style])

  return (
    <LibText testID={testID} style={textStyle} {...restProps}>
      {children}
    </LibText>
  )
})

export default Text
