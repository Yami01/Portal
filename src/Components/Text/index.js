/**
 *
 * @flow
 *
 */

import { FONT_BOLD, FONT_LIGHT, FONT_REGULAR, FONT_SEMIBOLD } from "./config"

export { default } from './Text'

export const fontWeightOptions = {
  bold: FONT_BOLD,
  light: FONT_LIGHT,
  semiBold: FONT_SEMIBOLD,
  regular: FONT_REGULAR,
}
