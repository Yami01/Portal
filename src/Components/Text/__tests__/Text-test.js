/**
 *
 * Text Test
 * @format
 * @flow
 *
 */

import React from 'react'
import renderer from 'react-test-renderer'
import Text from '../Text'
import { fontWeightOptions } from '../index'

const content = 'Testing Content'

describe('Text', () => {
  test('Text with normal content', () => {
    const tree = renderer.create(<Text>{content}</Text>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Text with bold font weight', () => {
    const tree = renderer
      .create(<Text fontWeight={fontWeightOptions.bold}>{content}</Text>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Text with bold font weight, font size, color', () => {
    const tree = renderer
      .create(
        <Text fontWeight={fontWeightOptions.bold} fontSize={24} color="blue">
          {content}
        </Text>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
