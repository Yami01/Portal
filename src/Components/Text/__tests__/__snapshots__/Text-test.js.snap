// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Text with bold font weight 1`] = `
<Text
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  Testing Content
</Text>
`;

exports[`Text Text with bold font weight, font size, color 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontSize": 24,
      },
      Object {
        "color": "blue",
      },
      undefined,
    ]
  }
>
  Testing Content
</Text>
`;

exports[`Text Text with normal content 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  Testing Content
</Text>
`;
